include/libdnf/rpm/transaction.hpp

```diff
@@ -0,0 +1,339 @@
+/*
+Copyright (C) 2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef LIBDNF_RPM_TRANSACTION_HPP
+#define LIBDNF_RPM_TRANSACTION_HPP
+
+#include "libdnf/rpm/package.hpp"
+#include "libdnf/utils/exception.hpp"
+#include "libdnf/utils/weak_ptr.hpp"
+
+#include <memory>
+
+namespace libdnf {
+
+class Base;
+
+}  // namespace libdnf
+
+namespace libdnf::rpm {
+
+class Transaction;
+class TransactionCB;
+class RpmProblemSet;
+
+/// Class represents one item in transaction set.
+class TransactionItem {
+public:
+    explicit TransactionItem(Package pkg) : pkg(std::move(pkg)) {}
+    const Package & get_pkg() const noexcept { return pkg; }
+    Package & get_pkg() noexcept { return pkg; }
+
+private:
+    Package pkg;
+};
+
+/// Class for access RPM header
+class RpmHeader {
+public:
+    std::string get_name() const;
+    uint64_t get_epoch() const noexcept;
+    std::string get_version() const;
+    std::string get_release() const;
+    std::string get_arch() const;
+    std::string get_evr() const;
+    std::string get_nevra() const;
+    std::string get_full_nevra() const;
+    unsigned int get_rpmdbid() const noexcept;
```

Comment:
I'm not sure if `get_rpmdbid()` is a good name for this function.
rpmdbid is frequently associated with sha1sum of a package and has (AFAIK) a completely different meaning
I'd rename it to `get_rpmdb_record_number()` which corresponds to rpm docstring: "@return rpmdb record number or 0"

Reply:
There is a method with the same name and result in the rpm::Package class. If you want to change the name of the method, I prefer change it in next PR. And on both places to stay consistent.